folsom-semantic-metrics/src/main/java/com/spotify/folsom/client/SemanticFolsomMetrics.java

```diff
@@ -0,0 +1,343 @@
+/*
+ * Copyright (c) 2014-2015 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.spotify.folsom.client;
+
+import com.codahale.metrics.Gauge;
+import com.codahale.metrics.Meter;
+import com.codahale.metrics.RatioGauge;
+import com.codahale.metrics.Timer;
+import com.spotify.folsom.GetResult;
+import com.spotify.folsom.MemcacheStatus;
+import com.spotify.folsom.Metrics;
+import com.spotify.metrics.core.MetricId;
+import com.spotify.metrics.core.SemanticMetricRegistry;
+
+import java.util.List;
+import java.util.concurrent.CompletionStage;
+
+/**
+ * {@link com.spotify.folsom.Metrics} implementation using semantic-metrics.
+ * Mostly a port of the YammerMetrics class.
```

Comment:
Aside, we should really rename that class, Yammer hasn't been involved for some time now.

Reply:
how about upgrading to the latest version of dropwizard-metrics and renaming the class to that? I can do that in another PR. 

Do you think its worth moving that class into its own module too?